test/object-store/c_api/c_api.cpp

```diff
@@ -4781,8 +4856,15 @@ static void realm_app_user2(void* p, realm_user_t* user, const realm_app_error_t
     }
 }
 
-TEST_CASE("C API app: link_user integration", "[c_api][sync][app]") {
-    TestAppSession session;
+TEST_CASE("C API app: link_user integration w/c_api transport", "[c_api][sync][app]") {
+    logger->set_level_threshold(realm::util::Logger::Level::TEST_ENABLE_SYNC_LOGGING_LEVEL);
```

Comment:
Tests that modify global state are problematic. You need to ensure that you reset the state back to its original value after testing, even if the test fails, otherwise you tests are no longer independant.

I suggest either moving all state (and helper functions as lambdas) into the test itself as private details for now. If you really intend for it to be shared, then try to initialize the state in the initializers prior to entering the test and not have the test modify the state.